pkg/frontend/transport/handler.go

```diff
@@ -41,6 +41,35 @@ var (
 	errRequestEntityTooLarge = httpgrpc.Errorf(http.StatusRequestEntityTooLarge, "http: request body too large")
 )
 
+const (
+	reasonRequestBodyTooLarge     = "request_body_too_large"
+	reasonResponseTooLarge        = "response_too_large"
+	reasonTooManyRequests         = "too_many_requests"
+	reasonTooLongRange            = "too_long_range"
+	reasonTooManySamples          = "too_many_samples"
+	reasonSeriesFetched           = "series_fetched"
+	reasonChunksFetched           = "chunks_fetched"
+	reasonChunkBytesFetched       = "chunk_bytes_fetched"
+	reasonDataBytesFetched        = "data_bytes_fetched"
+	reasonSeriesLimitStoreGateway = "series_limit_store_gateway"
+	reasonChunksLimitStoreGateway = "chunks_limit_store_gateway"
+	reasonBytesLimitStoreGateway  = "bytes_limit_store_gateway"
+)
+
+var (
+	LimitTooManySamples    = `query processing would load too many samples into memory`
```

Comment:
Does this have to be exported? I only see this being used in the `transport` package.